packages/std/examples/schema.rs

```diff
@@ -13,20 +13,37 @@ fn main() {
 
     export_schema(&schema_for!(Env), &out_dir);
     export_schema(&schema_for!(CosmosMsg), &out_dir);
-    export_schema(&schema_for!(InitResult), &out_dir);
-    export_schema(&schema_for!(HandleResult), &out_dir);
-    export_schema(&schema_for!(QueryResult), &out_dir);
+    export_schema_with_title(&mut schema_for!(InitResult), &out_dir, "InitResult");
+    export_schema_with_title(&mut schema_for!(HandleResult), &out_dir, "HandleResult");
+    export_schema_with_title(&mut schema_for!(QueryResult), &out_dir, "QueryResult");
 }
 
-/// Writes schema to file. Overwrites existing file.
-/// Panics on any error writing out the schema.
+// Exports a schema, auto-generating filename based on the metadata title of the generated schema.
 fn export_schema(schema: &RootSchema, out_dir: &PathBuf) -> () {
     let title = schema
         .schema
         .metadata
         .as_ref()
         .map(|b| b.title.clone().unwrap_or("untitled".to_string()))
         .unwrap_or("unknown".to_string());
+    write_schema(schema, out_dir, &title);
+}
+
+// use this if you want to override the auto-detected name of the object.
+// very useful when creating an alias for a type-alias.
+fn export_schema_with_title(schema: &mut RootSchema, out_dir: &PathBuf, title: &str) -> () {
```

Comment:
Can we create a mutable `.clone()` of `schema` to keep the caller code as sumple as possible?

Alternatively we can move the `RootSchema` by value also also avoid all the `&` in the caller code.

Reply:
I like pass by value. No need for ref